### DISCUSSION TITLE
Delete versions

### DISCUSSION THREAD

**Discussion by kohanla** [minio/minio#17011] [Q&A] [answered]
Hi,

I just started with Minio :-) Great product!

I setup a couple of buckets and replication. In order to have replications you need to enable versioning. So far so good. Everything works. But there is one issue. I would like to delete versions. I can list the versions with

`mc ls myminio/bucket -versions -r`

and I could delete them with

`mc rm -r --non-current --versions --force --dry-run myminio/bucket`

But I am wondering what I have to do in order to get "automatic deletion"? It seems I have to create a lifecycle rule for that.

Maybe something like that?

`mc ilm rule add  myminio/bucket --noncurrent-expire-days 1 --expire-delete-marker`

I waited for a couple of days ... But nothing has been deleted. I guess I probably misunderstood something in the documentation. What ilm rules I should I add in order to delete all "old" versions automatically (as fast as possible)?

**Comment by NianwenDan** [comment]
I am having similar issues right now. Did you find a way to solve the problem? I am not sure what exactly 1 day is based on. Is it the original file modification date or the deletion date?

**Comment by deajan** [comment]
Up, would love to know this too ;)

**Comment by tekkisse** [comment, marked as answer]
see : https://min.io/docs/minio/linux/administration/object-management/object-lifecycle-management.html  and second part of https://stackoverflow.com/questions/69031738/minio-is-it-possible-to-set-automatic-deletion-from-bucket-of-the-object-after-t
